cmd/juju/common/cloud.go

```diff
@@ -93,3 +93,27 @@ func BuiltInClouds() (map[string]jujucloud.Cloud, error) {
 	}
 	return allClouds, nil
 }
+
+// AnyCloudByName returns a cloud for given name
+// regardless of whether it's public, private or builtin cloud.
+// Since this method caters for builtin clouds, unlike cloud.CloudByName,
+// it should be used in cmd.
+func AnyCloudByName(cloudName string) (*jujucloud.Cloud, error) {
```

Comment:
It could but I did not want people to confuse it with cloud.CloudByName and wanted to make it absolutely, positively, blatantly obvious that one has a limitation over the other, especially when called from commands...
I am not too attached to the name if you insist though :D